packages/mui-material/src/ButtonGroup/ButtonGroup.js

```diff
@@ -252,7 +264,19 @@ const ButtonGroup = React.forwardRef(function ButtonGroup(inProps, ref) {
       ownerState={ownerState}
       {...other}
     >
-      <ButtonGroupContext.Provider value={context}>{children}</ButtonGroupContext.Provider>
+      <ButtonGroupContext.Provider value={context}>
+        {React.Children.map(children, (child, index) => {
+          if (!React.isValidElement(child)) {
+            return child;
+          }
```

Comment:
What case is this covering?

Reply:
Few use cases I could think of when the child is not a valid React element are:
```jsx
<ButtonGroup>
  <Button>Home</Button>
  <Button>About</Button>
  Some plain text here, which will be considered a non-valid element.
  <Button>Contact</Button>
</ButtonGroup>
```
Other one is suppose if a button is rendered conditionally:
```jsx
<ButtonGroup variant="outlined">{showButton ? <Button>Contact</Button> : null}</ButtonGroup>;
```
In such cases, we don't want to wrap the context provider on those child elements.